From 5613927a21b26768e421078dedc18be46e804ddd Mon Sep 17 00:00:00 2001
From: Mathieu Tortuyaux <mtortuyaux@microsoft.com>
Date: Thu, 23 Jul 2026 11:39:50 +0200
Subject: [PATCH] glib_stubs: undefine g_free

otherwise compiler tries to expand the macro to a function. Noticed with
glib >= 2.84.4

```
In file included from /build/amd64-usr/usr/include/glib-2.0/glib/glist.h:34,
                 from /build/amd64-usr/usr/include/glib-2.0/glib/ghash.h:36,
                 from /build/amd64-usr/usr/include/glib-2.0/glib.h:52,
                 from ../lib/rpcChannel/glib_stubs.c:28:
/build/amd64-usr/usr/include/glib-2.0/glib/gmem.h:172:27: error: expected declaration specifiers or '...' before '(' token
  172 |   (__builtin_object_size ((mem), 0) != ((size_t) - 1)) ?                       \
      |                           ^
../lib/rpcChannel/glib_stubs.c:38:6: note: in expansion of macro 'g_free'
   38 | void g_free(void *p) { free(p); }
      |      ^~~~~~
/build/amd64-usr/usr/include/glib-2.0/glib/gmem.h:172:34: error: expected declaration specifiers or '...' before numeric constant
  172 |   (__builtin_object_size ((mem), 0) != ((size_t) - 1)) ?                       \
      |                                  ^
../lib/rpcChannel/glib_stubs.c:38:6: note: in expansion of macro 'g_free'
   38 | void g_free(void *p) { free(p); }
      |      ^~~~~~
/build/amd64-usr/usr/include/glib-2.0/glib/gmem.h:172:37: error: expected ')' before '!=' token
  172 |   (__builtin_object_size ((mem), 0) != ((size_t) - 1)) ?                       \
      |                                     ^~
../lib/rpcChannel/glib_stubs.c:38:6: note: in expansion of macro 'g_free'
   38 | void g_free(void *p) { free(p); }
```

Signed-off-by: Mathieu Tortuyaux <mtortuyaux@microsoft.com>
---
 open-vm-tools/lib/rpcChannel/glib_stubs.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/open-vm-tools/lib/rpcChannel/glib_stubs.c b/open-vm-tools/lib/rpcChannel/glib_stubs.c
index c32deb073..5b3a76592 100644
--- a/open-vm-tools/lib/rpcChannel/glib_stubs.c
+++ b/open-vm-tools/lib/rpcChannel/glib_stubs.c
@@ -33,6 +33,8 @@
 #error "This file should be compiled for RPCI-only channel!"
 #endif
 
+#undef g_free
+
 void *g_malloc0(size_t s) { return Util_SafeCalloc(1, s); }
 void *g_malloc0_n(size_t n, size_t s) { return Util_SafeCalloc(n, s); }
 void g_free(void *p) { free(p); }
